### PR TITLE
feat(transloco): allow to provide multiple scopes via `provideTranslocoScopes`

### PR DESCRIPTION
**Pull request by EricPoul**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/transloco/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/ngneat/transloco/issues/708

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by EricPoul** [comment]
@NetanelBasal could someone check this one?

**Comment by alexfriesen** [comment]
This looks like a breaking change as you are removing `provideTranslocoScope`. I think adding `provideTranslocoScopes` as an additional function would be a better idea

**Comment by EricPoul** [comment]
I don't mind adding a new function and the previous one marking as deprecated

**Review by farshadhemmati — APPROVED** [review]

**Comment by EricPoul** [comment]
@farshadhemmati would you like to keep an old `provideTranslocoScope` in addition to `provideTranslocoScopes` or just a new `provideTranslocoScopes` with breaking changes?

File: libs/transloco/src/lib/transloco.providers.ts [hunk 1]
```diff
@@ -76,12 +76,12 @@ export function provideTranslocoLoader(loader: Type<TranslocoLoader>) {
   ]);
 }
 
-export function provideTranslocoScope(scope: TranslocoScope) {
-  return {
+export function provideTranslocoScopes(...scopes: TranslocoScope[]) {
```
Comment (shaharkazaz) [inline libs/transloco/src/lib/transloco.providers.ts:79]: keep the original name `provideTranslocoScope` and let's merge this, no need to make this a breaking change over a name as all the previous usages will still work as expected (correct my if I'm wrong)

**Comment by shaharkazaz** [comment]
@EricPoul We can merge this let's just skip the function rename

**Comment by majkers** [comment]
@EricPoul @shaharkazaz Is it just me or there is something wrong with it now?

I have:

```
@Component({
...
  providers: [provideTranslocoScope('ea')],
  standalone: true,
...
})
```

and then:

`private scope: TranslocoScope = inject(TRANSLOCO_SCOPE);
`
and when calling 

`this.translocoService.selectTranslate('header.logOut', undefined, this.scope)
`
I get an error because it is an array now:

```
core.mjs:6531 ERROR TypeError: lang.split is not a function
    at getLangFromScope (jsverse-transloco.mjs:379:17)
    at TranslocoService.isScopeWithLang (jsverse-transloco.mjs:658:28)
    at TranslocoService.selectTranslate (jsverse-transloco.mjs:642:39)
```

but when I go back to decalre scope like this:

it is working OK

```
providers: [{
    provide: TRANSLOCO_SCOPE,
    useValue: 'ea'
  }],
```

**Comment by shaharkazaz** [comment]
@majkers hm... It's a case that got missed when adding this feature, since now the scope is provided as multi the inject will return an array which I guess is a breaking change.

As a workaround you can just access the scope at position 0.

A proposed fix might be to drop the multi flag in case one scope is provided, but the thing with this is that you can get either an array or a scope if you inject it.

WDYT? @EricPoul

**Comment by shaharkazaz** [comment]
@majkers thanks for the extra info!
Would you like to open a PR for the fix? Or I'll take it?

**Comment by shaharkazaz** [comment]
@majkers There is always a first time! I think it's a nice chance to contribute to an open source, especially after you have already collected all the information needed 😀 

But of course it's your call 🙏

**Comment by shaharkazaz** [comment]
@majkers should be ok in v7.4.1

**Pull request closed without merging**